src/spring2024/s24_my_ptr.cpp

```diff
@@ -0,0 +1,250 @@
+#include <iostream>
+#include <memory>
+#include <utility>
+
+// This file contains codes used in the Spring2024 15445/645 C++ bootcamp.
+// It dives deeply into C++ new features like move constructor/assignment, move semantics, unique_ptr, shared_ptr,
+// wrapper class, etc., by implementing a simple version of unique_ptr from scratch.
+// **IMPORTANT**: please BEGIN reading from MAIN function!
+
+// Notice: it is just a simple version of std::unique_pointer<T>. The real implementation is much more complex!
```

Comment:
What is a simple version of std::unique_ptr? I'm assuming you mean "implementation below" or "our implementation here" or something similar/